docker-compose.yaml

```diff
@@ -0,0 +1,79 @@
+x-default: &node
+  image: ${CCAO_REGISTRY_URL}/service-spark-iasworld:latest
+  # Corresponds to the user set in the Dockerfile and the shiny-server
+  # user on the server for proper write perms to mounted directories
+  user: 1003:0
+  build:
+    context: .
+    dockerfile: Dockerfile
+  volumes:
+    - ./drivers:/jdbc:ro
+    - ./src:/tmp/src:ro
+    - ./config:/tmp/config:ro
+    - ./config/spark-defaults.conf:/opt/bitnami/spark/conf/spark-defaults.conf:ro
+    - ./target:/tmp/target:rw
+  secrets:
+    - IPTS_PASSWORD
+  networks:
+    - sparknet
+
+services:
+  spark-node-master:
+    <<: *node
+    container_name: spark-node-master
+    hostname: spark-node-master
+    environment:
+      - SPARK_MODE=master
+
+      - SPARK_PUBLIC_DNS=${SPARK_PUBLIC_IP}
+      - SPARK_RPC_AUTHENTICATION_ENABLED=no
+      - SPARK_RPC_ENCRYPTION_ENABLED=no
+      - SPARK_LOCAL_STORAGE_ENCRYPTION_ENABLED=no
+      - SPARK_SSL_ENABLED=no
+      - SPARK_USER=spark
+      - IPTS_HOSTNAME
+      - IPTS_PORT
+      - IPTS_SERVICE_NAME
+      - IPTS_USERNAME
+      - IPTS_TABLE
```

Comment:
These are shared by both containers, but unfortunately I don't think there's a way to dedupe them.